src/com/seafile/seadroid2/ui/activity/BrowserActivity.java

```diff
@@ -309,6 +305,81 @@ public void onPageScrolled(int arg0, float arg1, int arg2) {
         Intent monitorIntent = new Intent(this, FileMonitorService.class);
         startService(monitorIntent);
 
+        fetchServerInfo();
+    }
+
+    private void fetchServerInfo() {
+        if (isServerProEdition())
+            return;
+        else {
+            // hide Activity tab and search menu
+            adapter.hideActivityTab();
+            indicator.notifyDataSetChanged();
+            adapter.notifyDataSetChanged();
+            // hide search menu
+            if (menuSearch != null)
+                menuSearch.setVisible(false);
+        }
+
+        if (!Utils.isNetworkOn())
+            return;
+
+        ConcurrentAsyncTask.execute(new FetchServerInfoTask());
+    }
+
+    class FetchServerInfoTask extends AsyncTask<Void, Void, ServerInfo> {
+
+        @Override
+        protected ServerInfo doInBackground(Void... params) {
+            try {
+                return dataManager.getServerInfo();
+            } catch (IOException e) {
+                e.printStackTrace();
+            } catch (SeafException e) {
+                e.printStackTrace();
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+            return null;
+        }
+
+        @Override
+        protected void onPostExecute(ServerInfo serverInfo) {
+            if (serverInfo == null)
+                return;
+
+            if (serverInfo.isProEdition()) {
+                // hide Activity tab and search menu
+                adapter.unHideActivityTab();
+                indicator.notifyDataSetChanged();
+                adapter.notifyDataSetChanged();
+                // hide search menu
+                if (menuSearch != null)
+                    menuSearch.setVisible(true);
+            }
+
+            serverInfo.setUrl(account.getServer());
+            saveServerProEdition(serverInfo);
+        }
+    }
+
+    private void saveServerProEdition(ServerInfo serverInfo) {
+        if (!serverInfoList.contains(serverInfo))
+            serverInfoList.add(serverInfo);
+    }
+
+    private boolean isServerProEdition() {
+        if (serverInfoList.isEmpty()
+                || account == null)
+            return false;
+
+        for (ServerInfo si : serverInfoList) {
+            if (si.getUrl().equals(account.getServer()))
+                return si.isProEdition();
```

Comment:
What about two accounts on the same server? Better save a map `<AccountSignature, ServerInfo>`, instead of a list.